pkg/controller/grouppermission/grouppermission_controller_test.go

```diff
@@ -0,0 +1,84 @@
+package grouppermission
+
+import (
+	"context"
+	"testing"
+
+	operatorconfig "github.com/openshift/rbac-permissions-operator/config"
+	routev1 "github.com/openshift/rbac-permissions-operator/pkg/apis"
+	managedv1alpha1 "github.com/openshift/rbac-permissions-operator/pkg/apis/managed/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	client "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func reset(ctx context.Context, r *ReconcileGroupPermission) {
+	r.client.Delete(ctx, makeGroupPermission())
+}
+
+// create a GroupPermission object so we can resigter it in the fake client
+func makeGroupPermission() *managedv1alpha1.GroupPermission {
+	return &managedv1alpha1.GroupPermission{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      operatorconfig.OperatorName,
+			Namespace: operatorconfig.OperatorNamespace,
+		},
+		TypeMeta: metav1.TypeMeta{},
+	}
+}
+
+// create fake client to mock API calls
+func newTestReconciler() *ReconcileGroupPermission {
+	return &ReconcileGroupPermission{
+		client: client.NewFakeClient(),
+		scheme: nil,
```

Comment:
should this be "scheme: scheme.Scheme" ?

Reply:
hm not sure if that matters? I am registering the operator type with the runtime scheme and adding the api to the scheme in my test function. (before doing this I got an api error)